src/deployment/deploy.py

```diff
@@ -694,42 +689,48 @@ def set_instance_config(self) -> None:
             tenants.append(tenant)
         update_allowed_aad_tenants(config_client, tenants)
 
-    def create_eventgrid(self) -> None:
-        logger.info("creating eventgrid subscription")
-        src_resource_id = self.results["deploy"]["fuzz-storage"]["value"]
-        dst_resource_id = self.results["deploy"]["func-storage"]["value"]
-
+    def remove_eventgrid(self) -> None:
         credential = AzureCliCredential()
         client = StorageManagementClient(
             credential, subscription_id=self.get_subscription_id()
         )
-        event_subscription_info = EventSubscription(
-            destination=StorageQueueEventSubscriptionDestination(
-                resource_id=dst_resource_id, queue_name="file-changes"
-            ),
-            filter=EventSubscriptionFilter(
-                included_event_types=[
-                    "Microsoft.Storage.BlobCreated",
-                    "Microsoft.Storage.BlobDeleted",
-                ]
-            ),
-            retry_policy=RetryPolicy(
-                max_delivery_attempts=30,
-                event_time_to_live_in_minutes=1440,
-            ),
-        )
+        try:
+            storage_accounts = client.storage_accounts.list_by_resource_group(
+                self.resource_group
+            )
+            for storage_account in storage_accounts:
+                if storage_account.name.startswith("fuzz"):
+                    src_resource_id = storage_account.id
+
+            if not src_resource_id:
+                return
+        except ResourceNotFoundError:
+            return
 
         client = EventGridManagementClient(
             credential, subscription_id=self.get_subscription_id()
         )
-        result = client.event_subscriptions.begin_create_or_update(
-            src_resource_id, "onefuzz1", event_subscription_info
-        ).result()
-        if result.provisioning_state != "Succeeded":
-            raise Exception(
-                "eventgrid subscription failed: %s"
-                % json.dumps(result.as_dict(), indent=4, sort_keys=True),
-            )
+
+        old_subscription_name = "onefuzz1"
+
+        def old_subscription_exists() -> bool:
+            try:
+                client.event_subscriptions.get(src_resource_id, old_subscription_name)
```

Comment:
If `get()` ever stopped throwing an exception, and instead started returning `None` if `old_subscription_name` is absent, then this would start silently failing.

Maybe we should defensively do something like;
```
subscription = client.event_subscriptions.get(src_resource_id, old_subscription_name)
return bool(subscription)
```

Reply:
i expect any error other than `ResourceNotFoundError` to be raised "loudly" here

Reply:
Right, loud failures are what will happen right now, since we don't try to catch any exception but `ResourceNotFoundError`. We assume other error conditions will fail loudly as uncaught exceptions. That makes sense.

I'm just noting that:
1. Right now we implicitly treat any non-throwing call to `get()` as proof that the subscription exists.
2. So, the upstream library could break us in a way that would pass `mypy` checks: if they made `get()` return an `Optional[_]`, and started returning `None` in some situations (_instead_ of throwing), we'd never see it.

Arguably too defensive, but I was imagining it'd be a big pain to track down if it ever happened!